compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp

```diff
@@ -762,12 +762,12 @@ static bool hasUnfusableUseInDispatch(Value v, Operation *dispatchOp) {
   for (OpOperand &use : v.getUses()) {
     Operation *user = use.getOwner();
 
-    // Do not fuse `index_cast` operations if it is already an operand of the
-    // owner.
-    if (auto indexCastOp = v.getDefiningOp<arith::IndexCastOp>()) {
-      if (user == dispatchOp) {
-        return true;
-      }
+    // Do not fuse operations if they are already an operand of the
+    // owner and have an index return type as that means its a shape
+    // computation that needs to happen on the host.
+    auto op = v.getDefiningOp();
+    if (op && user == dispatchOp && v.getType().isIndex()) {
```

Comment:
For now just add a check that the `dispatchOp` is a `flow.dispatch.region`.

Reply:
Added the check despite your comment below that it is okay as is, as that check doesnt hurt. Can remove it though.